### PR TITLE
Rearranging which test configurations test JAX

### PR DESCRIPTION
**Pull request by DanPuzzuoli**
### Summary

Sets test environments to use the JAX version for ubuntu and macOS, and only run the non-jax test environments on windows.

### CODE REVIEW

**Review by chriseclectic — COMMENTED** [review]
If we are only going to have a single workflow for each OS (Linux+mac w jax, win wo jax) we could probably edit the workflow so that you just have a conditional for the os on which tox job to run for installing dependencies (@mtreinish probably knows syntax for doing that)

**Review by mtreinish — COMMENTED** [review]
Yeah you can do that. If you change:
```yaml
- name: Install and Run Tests
  run: tox -e py
```

to something like

```yaml
- name: Install and Run Tests
  run: tox -e jax
  if: runner.os != "Windows"
- name: Install and Run Tests
  run: tox -e py
  if: runner.os == "Windows"  
```

you can eliminate the second jax specific job and have a single job definition for the full matrix

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -7,6 +7,13 @@ on:
 jobs:
   tests:
     name: tests-python${{ matrix.python-version }}-${{ matrix.os }}
+    rules:
+      - if: ${{ matrix.os }} == "windows-latest"
+        variables:
+          run_tests_command: "tox -e py"
+      - if: ${{ matrix.os }} != "windows-latest"
+        variables:
+          run_tests_command: "tox -e jax"
```
Comment (mtreinish) [inline .github/workflows/main.yml:16]: I don't think `rules` is a valid key. Reading the env variable section of github actions's documentation I'm not sure there is a way to do conditional variables like this: https://docs.github.com/en/actions/learn-github-actions/environment-variables

There might be a way with inline expressions but that'll be a bit harder to read imo if possible. I think my earlier suggestion is the best path forward here. Where we just do an if else in the steps to run either job based on the runner

```suggestion
    env:
      run_tests_command: ${{ 
        variables:
          run_tests_command: "tox -e py"
      - if: ${{ matrix.os }} != "windows-latest"
        variables:
          run_tests_command: "tox -e jax"
```

Comment (DanPuzzuoli) [reply .github/workflows/main.yml]: Thanks, I had tried that but I think it wasn't quite right, and then got lost in some random pages through google :P

**Review by chriseclectic — APPROVED** [review]

**Pull request merged**